src/sdp/media/ersip_sdp_attr_rtpmap.erl

```diff
@@ -0,0 +1,209 @@
+%%% Types
+%%%===================================================================
+
+-record(rtpmap, {
+    type           :: payload_type(),
+    encoding_name  :: encoding_name(),
+    clock_rate     :: clock_rate(),
+    params         :: encoding_params() 
+}).
+
+-type payload_type()     :: binary().
+-type encoding_name()    :: binary().
+-type clock_rate()       :: pos_integer().
+-type encoding_params()  :: pos_integer() | undefined.
+-type rtpmap()           :: #rtpmap{}.
+-type parse_result()     :: parse_result(rtpmap()).
+-type parse_result(T)    :: ersip_parser_aux:parse_result(T).
+
+-export_type([
+    payload_type/0,
+    clock_rate/0,
+    encoding_name/0    
+]).
+
+%%%===================================================================
+%%% API
+%%%===================================================================
+
+-spec new(payload_type(), encoding_name(), clock_rate()) -> rtpmap().
+new(Type, Name, ClockRate) ->
+    new(Type, Name, ClockRate, undefined).
+
+-spec new(payload_type(), encoding_name(), clock_rate(), encoding_params()) -> rtpmap().
+new(Type, Name, ClockRate, Params) ->
+    #rtpmap{
+        type = Type,
+        encoding_name = Name,
+        clock_rate = ClockRate,
+        params = Params
+    }.
+
+-spec payload_type(rtpmap()) -> payload_type().
+payload_type(#rtpmap{type = Type}) ->
+    Type.
+
+-spec encoding_name(rtpmap()) -> encoding_name().
+encoding_name(#rtpmap{encoding_name = Name}) ->
+    Name.
+
+-spec clock_rate(rtpmap()) -> clock_rate().
+clock_rate(#rtpmap{clock_rate = ClockRate}) ->
+    ClockRate.
+
+-spec encoding_params(rtpmap()) -> encoding_params().
+encoding_params(#rtpmap{params = Params}) ->
+    Params.
+
+-spec set_payload_type(payload_type(), rtpmap()) -> rtpmap().
+set_payload_type(Type, Rtpmap) ->
+    Rtpmap#rtpmap{type = Type}.
+
+-spec set_encoding_name(encoding_name(), rtpmap()) -> rtpmap().
+set_encoding_name(Name, Rtpmap) ->
+    Rtpmap#rtpmap{encoding_name = Name}.
+
+-spec set_clock_rate(clock_rate(), rtpmap()) -> rtpmap().
+set_clock_rate(ClockRate, Rtpmap) ->
+    Rtpmap#rtpmap{clock_rate = ClockRate}.
+
+-spec set_encoding_params(encoding_params(), rtpmap()) -> rtpmap().
+set_encoding_params(Params, Rtpmap) ->
+    Rtpmap#rtpmap{params = Params}.
+
+-spec parse(binary()) -> parse_result().
+parse(Bin) ->
+    do_parse_rtpmap(Bin).
+
+-spec assemble(rtpmap()) -> iolist().
+assemble(Rtpmap) ->
+    assemble_rtpmap(Rtpmap).
+
+-spec assemble_bin(rtpmap()) -> binary().
+assemble_bin(Rtpmap) ->
+    iolist_to_binary(assemble(Rtpmap)).
+
+%%%===================================================================
+%%% Internal implementation
+%%%===================================================================
+%% https://datatracker.ietf.org/doc/html/rfc4566#section-6
+%%
+%% a=rtpmap:<payload type> <encoding name>/<clock rate> [/<encoding parameters>]
+%%
+
+-spec do_parse_rtpmap(binary()) -> parse_result().
+do_parse_rtpmap(Bin) ->
+    Parsers = [
```

Comment:
For parsing I would follow following syntax:
https://datatracker.ietf.org/doc/html/rfc8866#section-6.6
Note that we already have:
* ersip_sdp_aux:parse_token/1 for encoding name
* ersip_parser_aux:parse_non_neg_int/1 for non-negative integers
*  fun(Bin) -> ersip_parser_aux:parse_sep($/, Bin) end to parse "/"
So basically it can be:
```
Parsers = [
 fun ersip_parser_aux:parse_non_neg_int/1,
 fun ersip_parser_aux:trim_lws/1,
 fun ersip_sdp_aux:parse_token/1,
 fun(Bin) -> ersip_parser_aux:parse_sep($/, Bin) end,
 fun ersip_parser_aux:parse_non_neg_int/1,
 fun parse_encoding_params/1 
]
```

Reply:
Rewrite fun parse/1, using existing functions